app/src/main/java/com/hover/stax/transactions/StaxTransaction.kt

```diff
@@ -100,7 +100,7 @@ data class StaxTransaction(
     var accountName: String? = null
 
     constructor(data: Intent, action: HoverAction, contact: StaxContact?, context: Context) : this(
-        data.getStringExtra(TransactionContract.COLUMN_UUID)!!,
+        data.getStringExtra(TransactionContract.COLUMN_UUID) ?: "",
```

Comment:
This might cause issues since the UUID is a unique index. I don't think this is where the issue is, because the crash comes when loading from the Hover db, not saving to the Stax db. I can't actually tell what is null from the exception, but I think the Hover db is returning a null transaction for the given uuid rather than the uuid being null.

Reply:
So the fix should be - make the `transaction` nullable inside `hover db` - let the query return a null value?

This will prevent the crash from happening @davkutalek

Reply:
yeah, the issue is that the TransactionDao in the SDK is in java, so it is automatically nullable. So yeah, I guess the `TransactionRepo.loadFromHover` call needs to return a nullable Transaction

Reply:
Do we want to fix that in this fix, or next release? We will have to do a release on the library as well

Reply:
There is nothing to change in the library since it is java, TransactionRepo is part of Stax

Reply:
I missed the java part, my bad - makes sense now

Reply:
Fixed @davkutalek 

Ready for review